src/NuGet.Core/NuGet.Commands/PackCommandRunner.cs

```diff
@@ -794,7 +797,23 @@ private string GetOutputPath(PackageBuilder builder, bool symbols = false, NuGet
 
         public static string GetInputFile(PackArgs packArgs)
         {
-            IEnumerable<string> files = packArgs.Arguments != null && packArgs.Arguments.Any() ? packArgs.Arguments : Directory.GetFiles(packArgs.CurrentDirectory);
+            bool usingDirectory = false;
+            if (packArgs.Arguments?.Count() == 1)
+            {
+                string first = packArgs.Arguments.First();
+                if (Directory.Exists(first))
+                {
+                    if (string.IsNullOrEmpty(packArgs.OutputDirectory))
+                    {
+                        packArgs.OutputDirectory = packArgs.CurrentDirectory;
```

Comment:
The reason for the code is that most of pack assumes that the current working directory is where the project is located, but the nupkg file should be output to where the user was when the command was run.  It should internally change directories to the project directory, but it needs to remember where to output the file.  So if no OutputDirectory was passed in as an argument, I set the location where the user was when the command was run as the output directory and then internally to a change directory to the project.  If the code needed the output directory somewhere else, it would be set to the correct value.  It just wouldn't be null like it would otherwise be.